### DISCUSSION TITLE
Which_key and adding more actions under existing defaults

### DISCUSSION THREAD

**Discussion by RizaHKhan** [LunarVim/LunarVim#3179] [General]
How do I add more actions under existing `mappings`?

Ie, here is a default inside `lua/lvim/core/which-key.lua`:

```      
      s = {
        name = "Search",
        b = { "<cmd>Telescope git_branches<cr>", "Checkout branch" },
        c = { "<cmd>Telescope colorscheme<cr>", "Colorscheme" },
        f = { "<cmd>Telescope find_files<cr>", "Find File" },
        h = { "<cmd>Telescope help_tags<cr>", "Find Help" },
```

How do I add another (ie, `q`) inside the `s` name space in the config file?

I tried this and it did not work:

```
 lvim.builtin.which_key.mappings["s"] = {
   ["s"] = { "<cmd>Telescope live_grep<cr>", "String" }
 }

```

**Comment by mateusabelli** [comment]
Hello @RizaHKhan, I managed to get this working, the setup is almost the same as any which key mapping but you place the table that you want to insert into first before the key attribution. See my two examples for better understanding.

lvim.builtin.which_key.mappings.**s**["q"] You just add the key for the table you want to preserve instead of overwriting it as usual.

I'm inserting "z" into the "s" table defined in the core settings which will contain all the previous keymaps.

```
lvim.builtin.which_key.mappings.s["z"] = {
  "<cmd>echo 'hello there'<cr>", "Say hello"
}
```

In this example I create new nested options inside the s keymap 

```
lvim.builtin.which_key.mappings.s["q"] = {
  name = "+Custom Commands",
  a = { "<cmd>echo 'it works'<cr>", "My first custom command" },
  b = { "<cmd>echo 'it works again'<cr>", "My second custom command" },
}
```

This is how it looks like.

![image](https://user-images.githubusercontent.com/43862225/194727535-ddba27c0-ac17-424f-bc94-f0b9f08b008a.png)

**Reply by RizaHKhan** [reply]
This fixed my issue. Thank you for the insight!!

**Reply by danielo515** [reply]
Why do you need `lvim.builtin.which_key.mappings.s["z"]` rather than just `lvim.builtin.which_key.mappings.s.z` ?
